api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,34 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const accessToken = 'x-access-token';
+  if (!req.headers[accessToken]) {
+    throw new ClientError(401, 'authentication required');
+  }
+  const token = req.headers[accessToken];
+  try {
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
If you `catch` here, you need pass the error to `next()` or your client will hang. Try it for yourself, you'll see.